player/android/app/src/main/kotlin/dev/mydia/player/MainActivity.kt

```
@@ -66,6 +77,121 @@ class MainActivity : FlutterActivity() {
+     *
+     * Only decoders are considered, and a codec is reported only if some
+     * decoder claims its MIME type. Anything that throws is skipped rather than
+     * failing the whole probe: a single vendor codec with a broken capability
+     * table must not cost the device its entire profile.
+     */
+    private fun videoDecoderCapabilities(): List<Map<String, Any>> {
+        val best = mutableMapOf<String, MutableMap<String, Any>>()
+
+        val infos = try {
+            MediaCodecList(MediaCodecList.REGULAR_CODECS).codecInfos
+        } catch (error: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)